Expensify/App
# Update timezone to update if user has it set to automatic

cc: @iwiznia 

### Details
<Explanation of the change or anything fishy that is going on>

### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>
Fixes https://expensify.slack.com/archives/C01GTK53T8Q/p1617920244466600?thread_ts=1617784393.408700&cid=C01GTK53T8Q

### Tests
1. Open your developer console (opt + cmd + j for chrome) and go to the `Sources` tab
2. Set a breakpoint at ` if (_.isObject(timezone) && timezone.automatic && timezone.selected !== currentTimezone) {
` in `AuthScreens.js` 
<details closed>
<summary>Video to set breakpoint</summary>
<br>

https://user-images.githubusercontent.com/16747822/114248995-2019e980-994e-11eb-807c-c1610e583020.mov

</details>

3. Reload the page
4. When the point stops at the debugger, manually update your timezone by running the following in the console 
```JS 
timezone = {automatic: true, selected: "Africa/Tripoli"};
```
5. Continue execution 
6. Navigate to the `Application` tab the developer console, click on the value under `Local Storage` on the left hand side.
<details closed>
<summary>Video to find personalDetails</summary>
<br>
    
https://user-images.githubusercontent.com/16747822/114108464-fb5b3e80-9887-11eb-8106-e340bf8fd95f.mov

</details>

7. Search for `myPersonalDetails`, and click on the value. 
8. Confirm it's update to your current timezone. 

### QA Steps
Repeat test steps (Web only QA) 

### Tested On

### Screenshots
N/A no UI changes.
